group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1197,6 +1197,45 @@ private void throwIfClassicProtocolIsNotSupported(
         }
     }
 
+    /**
+     * Validates if the consumer group member uses the classic protocol.
+     *
+     * @param member The ConsumerGroupMember.
+     */
+    private void throwIfMemberDoesNotUseClassicProtocol(ConsumerGroupMember member) {
+        if (!member.useClassicProtocol()) {
+            throw new UnknownMemberIdException(
+                String.format("Member %s does not use the classic protocol.", member.memberId())
+            );
+        }
+    }
+
+    /**
+     * Validates if the generation id and the protocol type from the request match those of the consumer group.
+     *
+     * @param group                 The ConsumerGroup.
+     * @param member                The ConsumerGroupMember.
+     * @param requestGenerationId   The generation id from the request.
+     * @param requestProtocolType   The protocol type from the request.
+     * @param requestProtocolName   The protocol name from the request.
+     */
+    private void throwIfGenerationIdOrProtocolUnmatched(
```

Comment:
nit: It may be better to split this one into two methods. One to validate the generation. Another one to validate the protocol type and name.